File: 7-kyu/the-dropwhile-function/javascript/solution.js
const dropWhile = (arr, pred) => {
  const span = (array, predicate) => {
    let i = 0;
    while (i < array.length) {
      if (!predicate(array[i])) {
        break;
      }
      i++;
    }
    return [array.slice(i), array.slice(0, i)];
  };
  return span(arr, pred)[0];
};

File: 7-kyu/the-dropwhile-function/python/solution.py
def span(arr, pred):
    i = 0
    while i < len(arr):
        if not pred(arr[i]):
            break
        i += 1
    return [arr[:i + 1], arr[i:]]

def drop_while(arr, pred):
    return span(arr, pred)[1]
